File: src/main/java/xyz/jessyu/StoreToDB.java
package xyz.jessyu;

import com.mongodb.ConnectionString;
import com.mongodb.DuplicateKeyException;
import com.mongodb.MongoWriteException;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.result.InsertOneResult;
import org.bson.Document;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;


/**
 * <p>Store the posts into MongoDB</p>
 * <p>Usage: Call the insertManyPostToDB function</p>
 * */
public class StoreToDB {
    private static final String DB_URL = Settings.getDbUrl();
    private static final String DB_NAME = Settings.getDbName();
    private static final String DB_COLLECTION = Settings.getDbCollection();
    private static final MongoClient mongoClient = MongoClients.create(new ConnectionString(DB_URL));
    private static final Logger logger = LoggerFactory.getLogger(StoreToDB.class);

    public static void insertPostToDB(Document post) {
        logger.info("Inserting post into DB");
        MongoCollection<Document> collection = mongoClient.getDatabase(DB_NAME)
                .getCollection(DB_COLLECTION);
        try {
            collection.insertOne(post);
        } catch(MongoWriteException e) {
            logger.error("Duplicate key found in database");
            System.out.println("Duplicate key found in database");
        }
    }
}

File: src/python/xyz/jessyu/store_to_db.py
"""
MongoDB storage operations for rental posts.
"""
from pymongo import MongoClient
from pymongo.errors import DuplicateKeyError
import logging
from .settings import Settings


logger = logging.getLogger(__name__)


# Singleton MongoDB client
_mongo_client = None
_collection = None


def _get_collection():
    """Get or create MongoDB collection."""
    global _mongo_client, _collection
    
    if _collection is None:
        db_url = Settings.get_db_url()
        db_name = Settings.get_db_name()
        db_collection = Settings.get_db_collection()
        
        _mongo_client = MongoClient(db_url)
        _collection = _mongo_client[db_name][db_collection]
    
    return _collection


class StoreToDB:
    """Handle storing rental posts to MongoDB."""
    
    @staticmethod
    def insert_post_to_db(post: dict):
        """
        Insert a single post into MongoDB.
        
        Args:
            post: Dictionary containing post data
        """
        logger.info("Inserting post into DB")
        
        try:
            collection = _get_collection()
            collection.insert_one(post)
            logger.info("Post inserted successfully")
        except DuplicateKeyError:
            logger.error("Duplicate key found in database")
        except Exception as e:
            logger.error(f"Error inserting post to database: {e}")
